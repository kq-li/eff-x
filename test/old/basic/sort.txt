_ : unit !write = store 0 5;
_ : unit !write = store 1 3;
_ : unit !write = store 2 9;
_ : unit !write = store 3 6;
_ : unit !write = store 4 1;
_ : unit !write = store 5 2;

i : int = 0;
while (lt i 6) {
  x_i : int !read = load i;
  j : int = sub i 1;
  while (ge j 0) {
    x_j : int !read = load j;
    if (lt x_i x_j) {
      _ : unit !write = store (add j 1) x_j;
      _ : unit !write = store j x_i;
    }
    j : int = sub j 1;
  }
  i : int = add i 1;
}

i : int = 0;
while (lt i 6) {
  _ : unit !output !read = print (load i);
  i : int = add i 1;
}
