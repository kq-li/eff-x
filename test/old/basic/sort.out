1
2
3
5
6
9
statements executed: 181
